lms/services/blackboard_api/client.py

```diff
@@ -186,3 +160,84 @@ def course_groups(
                 raise ExternalAsyncRequestError(response=response)
 
         return self_enrollment_groups + instructor_only_groups
+
+    def _list_files_url(self, course_id, folder_id=None) -> str:
+        """Return the url for the list files API endpoint."""
+        path = f"courses/uuid:{course_id}/resources"
+
+        if folder_id:
+            # Get the files and folders in the given folder instead of the
+            # course's top-level files and folders.
+            path += f"/{folder_id}/children"
+
+        return (
+            path
+            + "?"
+            + urlencode(
+                {
+                    "limit": PAGINATION_LIMIT,
+                    "fields": "id,name,type,modified,mimeType,size,parentId",
+                }
+            )
+        )
+
+    def _store_files(self, course_id, files: List[dict]):
+        """Store the given files in the DB."""
+        self._file_service.upsert(
+            [
+                {
+                    "type": "blackboard_file"
+                    if file["type"] == "File"
+                    else "blackboard_folder",
+                    "course_id": course_id,
+                    "lms_id": file["id"],
+                    "name": file["name"],
+                    "size": file["size"],
+                    "parent_lms_id": file["parentId"],
+                }
+                for file in files
+            ]
+        )
+
+    def _get_all_following_pages(self, response, schema) -> List:
+        """
+        Get all the paginated responses to `path`.
+
+        Return the results of all of them in one list after parsing them with `schema`.
+        """
+        results = schema(response).parse()
+        for _ in range(PAGINATION_MAX_REQUESTS - 1):
+            path = response.json().get("paging", {}).get("nextPage")
+            if not path:
+                break
+
+            response = self._api.request("GET", path)
+            results.extend(schema(response).parse())
+
+        return results
```

Comment:
Ok, so this is pagination code that was already here, but it's just been moved out into a separate method?